src/system_data.jl

```diff
@@ -780,11 +780,11 @@ function deserialize(
             attributes_by_uuid[uuid] = attr
         end
     end
-    for component in raw["components"]
-        if haskey(component, "attributes_container")
-            component["attributes_container"] = deserialize(
+    for component in Iterators.Flatten((raw["components"], raw["masked_components"]))
+        if haskey(component, "supplemental_attributes_container")
```

Comment:
One potential problem: this requires that every component that wants supplemental attributes has to use this field name. What do you think about such a requirement?

This solution is advantageous because we can deserialize the attributes here without inspecting types. Another solution is to determine the real types here. That _might_ slow things down some. Another solution is to deserialize attributes in the parent package (PowerSystems). We could think more and test performance impacts.